pop-api/examples/fungibles/lib.rs

```diff
@@ -1,108 +1,268 @@
+		/// * - `token` - The token.
 		#[ink(constructor, payable)]
-		pub fn new() -> Self {
-			Default::default()
+		pub fn existing(id: TokenId) -> Result<Self, Psp22Error> {
+			// Make sure token exists.
+			if !api::token_exists(id).unwrap_or_default() {
+				return Err(Psp22Error::Custom(String::from("Token does not exist")));
+			}
+			Ok(Self { id })
 		}
 
-		#[ink(message)]
-		pub fn total_supply(&self, token: TokenId) -> Result<Balance> {
-			api::total_supply(token)
+		/// Instantiate the contract and create a new token. The token identifier will be stored
+		/// in contract's storage.
+		///
+		/// # Parameters
+		/// * - `id` - The identifier of the token.
+		/// * - `min_balance` - The minimum balance required for accounts holding this token.
+		// The `min_balance` ensures accounts hold a minimum amount of tokens, preventing tiny,
+		// inactive balances from bloating the blockchain state and slowing down the network.
+		#[ink(constructor, payable)]
+		pub fn new(id: TokenId, min_balance: Balance) -> Result<Self, Psp22Error> {
+			let instance = Self { id };
+			let contract_id = instance.env().account_id();
+			api::create(id, contract_id, min_balance).map_err(Psp22Error::from)?;
+			instance
+				.env()
+				.emit_event(Created { id, creator: contract_id, admin: contract_id });
+			Ok(instance)
 		}
+	}
 
+	impl Psp22 for Fungible {
+		/// Returns the total token supply.
 		#[ink(message)]
-		pub fn balance_of(&self, token: TokenId, owner: AccountId) -> Result<Balance> {
-			api::balance_of(token, owner)
+		fn total_supply(&self) -> Balance {
+			api::total_supply(self.id).unwrap_or_default()
 		}
 
+		/// Returns the account balance for the specified `owner`.
+		///
+		/// # Parameters
+		/// - `owner` - The account whose balance is being queried.
 		#[ink(message)]
-		pub fn allowance(
-			&self,
-			token: TokenId,
-			owner: AccountId,
-			spender: AccountId,
-		) -> Result<Balance> {
-			api::allowance(token, owner, spender)
+		fn balance_of(&self, owner: AccountId) -> Balance {
+			api::balance_of(self.id, owner).unwrap_or_default()
 		}
 
+		/// Returns the allowance for a `spender` approved by an `owner`.
+		///
+		/// # Parameters
+		/// - `owner` - The account that owns the tokens.
+		/// - `spender` - The account that is allowed to spend the tokens.
 		#[ink(message)]
-		pub fn transfer(&mut self, token: TokenId, to: AccountId, value: Balance) -> Result<()> {
-			api::transfer(token, to, value)
+		fn allowance(&self, owner: AccountId, spender: AccountId) -> Balance {
+			api::allowance(self.id, owner, spender).unwrap_or_default()
 		}
 
+		/// Transfers `value` amount of tokens from the caller's account to account `to`
+		/// with additional `data` in unspecified format.
+		///
+		/// # Parameters
+		/// - `to` - The recipient account.
+		/// - `value` - The number of tokens to transfer.
+		/// - `data` - Additional data in unspecified format.
 		#[ink(message)]
-		pub fn transfer_from(
+		fn transfer(
```

Comment:
Sorry but this function doesnt make any sense to me. 

- Someone calls it specifying a beneficiary.
- The contract then calls the api to transfer tokens that it holds (contract is the origin) to the beneficary.
- It then emits an event saying that the transfer occurred and that the from was the caller when in fact it was the contract.

Anyone can call this to drain the contract and it is therefore a bad example. It should be using `transfer_from` imo, where the caller has pre-approved the contract to be able to spend some allowance by registering as much with the fungibles/assets pallet.